.github/workflows/test_deploy.yml

```diff
@@ -629,15 +626,17 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
       - name: Install dependencies
+        env:
+          POETRY_VIRTUALENVS_CREATE: false
         run: |
           brew update || true
           brew unlink git-lfs || true
           curl -L https://raw.githubusercontent.com/Homebrew/homebrew-core/43842898fd3ff43273466052722f5ba2789196cb/Formula/git-lfs.rb > git-lfs.rb && brew install git-lfs.rb && rm git-lfs.rb
           brew install shellcheck node || brew link --overwrite node
           python -m pip install --upgrade pip
-          python -m pip install wheel
+          python -m pip install wheel poetry
```

Comment:
Aren't `coveralls`, `twine` and `poetry-dynamic-versioning-plugin` needed here?

Reply:
We only run coveralls on linux (no point checking on mac as well). 
In the linux steps, we just include twine so the install is the same in all steps, because of the dependency cache (all installs need to be identical), but on the mac action we don't use a cache, so it's not needed